### PR TITLE
Allow duplicate uniqueKeyPolicy in DocumentCollection

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
# Description

This PR address a customer-reported issue of a transient error - I haven't been able to reproduce this myself - but the change is extremely low risk and can help making the Java SDK/Spark connector more robust.

## Reported exception:

```java
2024-08-19 12:11:41,861 ERROR Executor [Executor task launch worker for task 3.0 in stage 26.0 (TID 23809)]: Exception in task 3.0 in stage 26.0 (TID 23809)
java.lang.IllegalArgumentException: Unable to parse JSON {"id":"COLLECTIONNAME","indexingPolicy":{"indexingMode":"consistent","automatic":true,"includedPaths":[{"path":"/*"}],"excludedPaths":[{"path":"/\"_etag\"/?"}],"spatialIndexes":[{"path":"/*","types":["Point","LineString","Polygon","MultiPolygon"]}]},"partitionKey":{"paths":["/pk"],"kind":"Hash","systemKey":false},"defaultTtl":2764800,"conflictResolutionPolicy":{"mode":"LastWriterWins","conflictResolutionPath":"/_ts","conflictResolutionProcedure":""},"geospatialConfig":{"type":"Geography"},"_rid":"RID","_ts":1587070752,"_self":"dbs/DBRID/colls/COLLECTIONRID/","_etag":"\"ETAG\"","_docs":"docs/","_sprocs":"sprocs/","_triggers":"triggers/","_udfs":"udfs/","_conflicts":"conflicts/","uniqueKeyPolicy":{},"uniqueKeyPolicy":{}}
	at azure_cosmos_spark.com.azure.cosmos.implementation.JsonSerializable.fromJson(JsonSerializable.java:645)
	at azure_cosmos_spark.com.azure.cosmos.implementation.JsonSerializable.<init>(JsonSerializable.java:84)
	at azure_cosmos_spark.com.azure.cosmos.implementation.Resource.<init>(Resource.java:83)
	at azure_cosmos_spark.com.azure.cosmos.implementation.DocumentCollection.<init>(DocumentCollection.java:81)
	at azure_cosmos_spark.com.azure.cosmos.models.CosmosContainerProperties.<init>(CosmosContainerProperties.java:66)
	at azure_cosmos_spark.com.azure.cosmos.models.CosmosContainerProperties$1.create(CosmosContainerProperties.java:414)
	at azure_cosmos_spark.com.azure.cosmos.SparkBridgeInternal$.getContainerPropertiesFromCollectionCache(SparkBridgeInternal.scala:102)
	at com.azure.cosmos.spark.CosmosWriterBase.<init>(CosmosWriterBase.scala:58)
	at com.azure.cosmos.spark.CosmosWriter.<init>(CosmosWriter.scala:34)
	at com.azure.cosmos.spark.ItemsDataWriteFactory.createWriter(ItemsDataWriteFactory.scala:51)
	at org.apache.spark.sql.execution.datasources.v2.DataWritingSparkTask$.run(WriteToDataSourceV2Exec.scala:439)
	at org.apache.spark.sql.execution.datasources.v2.V2TableWriteExec.$anonfun$writeWithV2$2(WriteToDataSourceV2Exec.scala:390)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:136)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:548)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1504)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:551)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
```
The duplicated **`"uniqueKeyPolicy":{},"uniqueKeyPolicy":{}`** is unexpected.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by tvaron3 — APPROVED** [review]
LGTM

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 #### Bugs Fixed
 
 #### Other Changes
+* Fixed an issue to avoid transient `IllegalArgumentException` due to duplicate json properties for the `uniqueKeyPolicy` property. - See [PR 41348](https://github.com/Azure/azure-sdk-for-java/pull/41348)
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:12]: wrong PR reference?

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

File: sdk/cosmos/azure-cosmos/CHANGELOG.md (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md]: Yeah - forgot to push my last commit. Fixed.

**Pull request merged**